### PR TITLE
POC: Theme Singleton for color definitions and UI toggling

### PR DESCRIPTION
**Pull request by jarolrod**
based on #101, #102, #106
This introduces a `Theme` singleton which contains color definitions so that colors can be used by name instead of by hex. Additionally, it adjusts color definitions depending on if the gui is in dark or light mode. Adds a `ThemeToggleButton` so that we can toggle between light and dark. Uses the `moon` and `sun` icon from [BitcoinIcons](https://github.com/BitcoinDesign/Bitcoin-Icons).

Color definitions are based on our color palettes as defined in our [design system](https://www.figma.com/file/GaCoOSNHB2yMB9ThiDtred/Bitcoin-Core-App-Main?node-id=2643%3A83514): 
color-definitions

Current views are:
light-toggle

dark-toggle

[Windows](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/win64/insecure_win_gui.zip?branch=pull/109)
[macOS](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos/insecure_mac_gui.zip?branch=pull/109)
[Android](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/android/insecure_android_apk.zip?branch=pull/109)

### CODE REVIEW

File: src/qml/controls/Theme.qml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+pragma Singleton
+import QtQuick 2.15
+
+QtObject {
+    property bool dark: false
+    property QtObject color: QtObject {
```
Comment (promag) [inline src/qml/controls/Theme.qml:6]: 2b2edac044bb68511558d91464608fb831b5816b

I think we could avoid so many ternaries:
```qml
readonly property ColorSet color: dark ? darkColorSet : lightColorSet

component ColorSet: QtObject {
    required property color background
    // ...
}

ColorSet {
    id: darkColorSet
    background: "black"
    // ...
}

ColorSet {
    id: lightColorSet
    background: "white"
    // ...
}
```

**Comment by promag** [comment]
@GBKS yes, like `color`, it can have more properties.

**Comment by jarolrod** [comment]
Replaced by #114

File: src/qml/controls/Theme.qml (hunk 1 above)
Comment (jarolrod) [reply src/qml/controls/Theme.qml:6]: Picked up in https://github.com/bitcoin-core/gui-qml/pull/168

**Pull request closed without merging**